linuxprivchecker.sh

```diff
@@ -1,252 +1,251 @@
 
-if [ $PKGMNGR -eq 1 ]; then
-    	formatCommand "dpkg -l | grep -iE '${EXTDGREP}'"
-elif [ $PKGMNGR -eq 2 ]; then
-	formatCommand "dnf -qa | grep -iE '${EXTDGREP}'"
-elif [ $PKGMNGR -eq 3 ]; then
-	formatCommand "rpm -qa | grep -iE '${EXTDGREP}'"
-fi
+  systemNAME="Bash version";
+  cmdRESPONSE "bash --version | grep version";
 
-echo -ne "\n${SECTION_LINE}\n"
-echo -e "[*] ENUMERATING INSTALLED LANGUAGES/TOOLS FOR SPLOIT BUILDING..."
+  systemNAME="Sudo version";
+  cmdRESPONSE "sudo -V | grep version";
 
-echo -e "\n[+] Installed Tools"
-formatCommand "which awk perl python ruby gcc cc vi vim nmap find netcat nc wget tftp ftp 2>/dev/null"
+  systemNAME="Apache Version and Modules";
+  cmdRESPONSE "apache2 -v 2>/dev/null; apache2ctl -M 2>/dev/null; httpd -v 2>/dev/null; apachectl -l";
 
-echo -e "\n[+] Related Shell Escape Sequences"
-if [ -x "$(command -v vi)" ]; then
-	formatCommand "echo -ne \"vi-->\t:!bash\n\""
-	formatCommand "echo -ne \"vi-->\t:set shell=/bin/bash:shell\n\""
-fi
+  systemNAME="Apache Config File";
+  cmdRESPONSE "cat /etc/apache2/apache2.conf";
 
-if [ -x "$(command -v vim)" ]; then
-	echo -ne "vim-->\t:!bash\n" | sed 's|^|    |'
-	echo -ne "vim-->\t:set shell=/bin/bash:shell\n" | sed 's|^|    |'
-fi
+  systemNAME="Processes and Packages Running as Root or other Superuser";
+  EXTDGREP="($(ps -u 0 | tail -n+2 | rev | cut -d " " -f 1 | rev | cut -d "/" -f1 | sort | uniq | xargs | tr " " "|"))";
+  cmdRESPONSE "if [ -x "$(command -v dpkg)" ]; then dpkg -l | grep -iE '${EXTDGREP}'; elif [ -x "$(command -v dnf)" ]; then dnf -qa | grep -iE '${EXTDGREP}'; elif [ -x "$(command -v rpm)" ]; then rpm -qa | grep -iE '${EXTDGREP}'; fi";
 
-if [ -x "$(command -v awk)" ]; then
-	echo -ne "awk-->\tawk 'BEGIN {system(\"/bin/bash\")}'\n" | sed 's|^|    |'
-fi
+  systemNAME="Installed Tools";
+  cmdRESPONSE "which awk perl python ruby gcc cc vi vim nmap find netcat nc wget tftp ftp";
 
-if [ -x "$(command -v perl)" ]; then
-	echo -ne "perl-->\tperl -e 'exec \"/bin/bash\";'\n" | sed 's|^|    |'
-fi
+  systemNAME="Related Shell Escape Sequences";
+  cmdRESPONSE "if [ -x "$(command -v vi)" ]; then echo -ne \"vi-->\t:!bash\n\"; echo -ne \"vi-->\t:set shell=/bin/bash:shell\n\"; fi; if [ -x "$(command -v vim)" ]; then echo -ne \"vim-->\t:!bash\n\" | sed 's|^|    |'; echo -ne \"vim-->\t:set shell=/bin/bash:shell\n\" | sed 's|^|    |'; fi; if [ -x "$(command -v awk)" ]; then echo -ne \"awk-->\tawk 'BEGIN {system(\"/bin/bash\")}'\n\" | sed 's|^|    |'; fi; if [ -x "$(command -v perl)" ]; then echo -ne \"perl-->\tperl -e 'exec \"/bin/bash\";'\n\" | sed 's|^|    |'; fi; if [ -x "$(command -v python)" ]; then echo -ne \"python-->\tpython -c '__import__(\"os\").system(\"/bin/bash\")'\n\" | sed 's|^|    |'; fi; if [ -x "$(command -v find)" ]; then echo -ne \"find->\tfind / -exec /usr/bin/awk 'BEGIN {system(\"/bin/bash\")}' \\;\n\" | sed 's|^|    |'; fi; if [ -x "$(command -v nmap)" ]; then echo -ne \"nmap-->\t--interactive\n\" | sed 's|^|    |'; fi";
```

Comment:
This is just a lot to consume. It's not unclear, but quite a bit a first glance. I imagine it was done this way to make it easy to pass to cmdRESPONSE

Reply:
@5aru No worries, I'm not in a rush here....so take your time...no one is going to die :)

And yes, its so that it can get passed into that cmdRESPONSE and not repeat the Related Shell Escape Sequences for each.
The code is the same, I just push it all in one line and added \ for the double quotations.

Reply:
oh man, that's ugly. Is there anyway to make that multiple lines without having to break it up? I would like for this script to be easy to read. Personally I think if its easy to read even for people who don't know bash that will make the credibility of the script to improve. I know a couple pen testers who don't know code and a complicated line like this would definetly turn them away. However if we can't split it up without a ton of encoding, then this will have to work.

Reply:
Hey @5aru 
Thanks for the feedback and I just want to make sure that you take your time, I do this because I enjoy it and want to learn more from others like you and @linted 

I totally agree with you, [-] and [+] are unwritten standards...maybe I can make some adjustments to the text right next to them so that they look cleaner.
I'll make some adjustments.

I see your points of not porting that over, the need to keep things to look like the initial python script and I failed to remember that.  I'll try and remember that when coding, and maybe assist updating the python script.

Line 215 code is good but its a bit scary because it has five different if statements written in one line.
Let me see what I can do to make it readable.

I finished my exam but I know I failed it because I think I scored ~40-50.  I learned a lot from it and really glad I failed because now I know what to work on.